middleware/service/unittest/source/manager/test_manager.cpp

```diff
@@ -1143,5 +1143,140 @@ namespace casual
          }
       }
 
+      TEST( service_manager, terminate_service_times_out__callee_acks_before_SIGTERM__expect_busy_before_terminated)
+      {
+         common::unittest::Trace trace;
+
+         constexpr auto configuration = R"(
+domain:
+   services:
+      -  name: "a"
+         execution:
+            timeout:
+               duration: "1ms"
+               contract: "terminate"
+)";
+
+         constexpr auto service = "a";
+
+         auto domain = local::domain( configuration);
+
+         service::unittest::advertise( { service});
+
+         // We reserve our service
+         {
+            auto lookup = common::service::Lookup{ service}();
+            EXPECT_TRUE( lookup.state == decltype( lookup.state)::idle);
+         }
+
+         // We sleep, expecting to be interrupted by the DM trying to murder us
+         // Note that in the scenario being tested here the callee is terminated after it acks, but we have to do it this way since we're both caller and callee
+         try
+         {
+            common::process::sleep( std::chrono::seconds{ 10});
+         }
+         catch( ...)
+         {
+            auto error = common::exception::capture();
+            EXPECT_TRUE( error.code() == common::code::signal::terminate);
+         }
+
```

Comment:
Hmm... This got me thinking... `terminate` is by definition not lethal. 
Are we thinking about this problem the wrong way?... 

This patch assumes that if a _lethal_ assassination is sent, we'll wait for the process::Exit event. This might not happen... 

I've got second thoughts about this... If this patch does not fix the problem 100% then we shouldn't apply it. As I see it the patch might make it worse, at least for `terminate`.

Your thoughts @AlbsAgren?

Reply:
Since this issue is a pretty extreme edge-case to begin with I'm inclined to agree. A bigger rethinking of how we handle assassinations in 1.7 might be a better way to go...

Reply:
I guess the alternative would be to 'rehabilitate' instances that get SIGTERM and then ACK. Though not super elegant...